src/main/java/gov/nasa/pds/tools/validate/SpecialConstantChecker.java

```diff
@@ -0,0 +1,172 @@
+package gov.nasa.pds.tools.validate;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import gov.nasa.arc.pds.xml.generated.SpecialConstants;
+import gov.nasa.pds.tools.validate.content.ProblemReporter;
+import gov.nasa.pds.tools.validate.content.SpecialConstantBitPatternTransforms;
+
+public class SpecialConstantChecker {
+  /**
+   * Use this when the special constant may not conform to the type constraints
+   * meaning only string comparisons can be done.
+   * @param value
+   * @param constants
+   * @return
+   */
+  public static boolean isNonConformantSpecialConstant(String value, SpecialConstants constants) {
```

Comment:
@al-niessner so per the [PDS4 IM Spec](https://pds.nasa.gov/datastandards/documents/im/v1/index_1L00.html), these are the special constants that do not require matching data types:
* error_constant - https://pds.nasa.gov/datastandards/documents/im/v1/index_1L00.html#attribute_pds_special_constants_pds_error_constant
* invalid_constant - https://pds.nasa.gov/datastandards/documents/im/v1/index_1L00.html#attribute_pds_special_constants_pds_invalid_constant
* missing_constant - https://pds.nasa.gov/datastandards/documents/im/v1/index_1L00.html#attribute_pds_special_constants_pds_missing_constant
* not_applicable_constant - https://pds.nasa.gov/datastandards/documents/im/v1/index_1L00.html#attribute_pds_special_constants_pds_not_applicable_constant
* saturated_constant - https://pds.nasa.gov/datastandards/documents/im/v1/index_1L00.html#attribute_pds_special_constants_pds_saturated_constant
* unknown_constant - https://pds.nasa.gov/datastandards/documents/im/v1/index_1L00.html#attribute_pds_special_constants_pds_unknown_constant

The rest all state that they must match the data type of the object and should throw an error if they do not match.

Reply:
done